File: with-lombok/src/test/java/TestValVar.java
import static org.assertj.core.api.Assertions.assertThat;

import java.util.List;
import java.util.Map;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import lombok.val;

class TestValVar {

    @Test
    @DisplayName("should test the lombok val operator")
    void shouldTestValOperator() {
        // final Map<String,List<String>>
        val complexeValue = Map.of(
                "firstList",
                List.of("one", "two", "three")
        );
        assertThat(complexeValue)
                .isNotEmpty()
                .isInstanceOf(Map.class);
        assertThat(complexeValue.get("firstList"))
                .isNotEmpty()
                .isInstanceOf(List.class);
        /*
         Forbidden: complexeType is final
         complexeType = Map.of(
              "secondList",
              List.of("one", "two")
         );
        */
    }

    @Test
    @DisplayName("should test the Java 10 var operator")
    void shouldTestVarOperator() {
        // non-final Map<String,List<String>>
        // *Non-lombok feature, using Java 10 var key declaration*
        var complexeVariable = Map.of(
                "thirdList",
                List.of("one", "two")
        );
        complexeVariable = Map.of();

        assertThat(complexeVariable).isEmpty();
    }
}

File: with-kotlin/src/test/kotlin/TestValVar.kt
import org.junit.jupiter.api.Test
import kotlin.test.assertTrue

class TestValVar {

    @Test
    fun `should test Kotlin val operator`() {
        val complexeValue = mapOf(
            "firstList" to listOf("one", "two", "three")
        )

        assertTrue(complexeValue.isNotEmpty())
        assertTrue(complexeValue is Map<String, List<String>>)

        /*
        Forbidden: value is not reassignable
        complexeValue = mapOf("secondList" to listOf("one", "two"))
         */
    }

    @Test
    fun `should test Kotlin var operator`() {
        var complexeVariable = mapOf(
            "thirdList" to listOf("one", "two", "three")
        )
        complexeVariable = emptyMap()

        assertTrue(complexeVariable.isEmpty())
        assertTrue(complexeVariable is Map<String, List<String>>)
    }
}
